lib/galaxy/config/__init__.py

```diff
@@ -1227,6 +1230,11 @@ def _set_galaxy_infrastructure_url(self, kwargs):
         if "UWSGI_PORT" in self.galaxy_infrastructure_url:
             raise Exception("UWSGI_PORT is not supported anymore")
 
+    def _set_email_from(self):
+        if not self.email_from:
+            hostname = self.galaxy_infrastructure_url or socket.getfqdn()
+            self.email_from = f"{DEFAULT_EMAIL_FROM_LOCAL_PART}@{hostname}"
```

Comment:
```suggestion
        if self.galaxy_infrastructure_url_set:
            self.hostname = urlparse(self.galaxy_infrastructure_url).hostname
        else:
            self.hostname = socket.getfqdn()
        if not self.email_from:
            self.email_from = f"{DEFAULT_EMAIL_FROM_LOCAL_PART}@{self.hostname}"
```

with a `from urllib.parse import urlparse` added at the top.